chalk-recursive/src/fixed_point.rs

```diff
@@ -0,0 +1,215 @@
+use crate::cache::Cache;
+use crate::search_graph::DepthFirstNumber;
+use crate::search_graph::SearchGraph;
+use crate::stack::{Stack, StackDepth};
+use crate::Minimums;
+use std::fmt::Debug;
+use std::hash::Hash;
+use tracing::debug;
+use tracing::{info, instrument};
+
+pub(super) struct RecursiveContext<K, V>
+where
+    K: Hash + Eq + Debug + Clone,
+    V: Debug + Clone,
+{
+    stack: Stack,
+
+    /// The "search graph" stores "in-progress results" that are still being
+    /// solved.
+    search_graph: SearchGraph<K, V>,
+
+    /// The "cache" stores results for goals that we have completely solved.
+    /// Things are added to the cache when we have completely processed their
+    /// result.
+    cache: Option<Cache<K, V>>,
+
+    /// The maximum size for goals.
+    max_size: usize,
+}
```

Comment:
Where is this used? This doesn't seem like the right place for this logically...

Reply:
The field is used to truncate goals and things.

Hmm, you're not wrong. We should eventually factor `RecursiveContext` into "local state" and "global state" -- the cache and max-size are both "global state". The "stack" and "search graph" are "local state" (there could be multiple copies of them).

Reply:
Well, truncation doesn't feel like logic that the "fixed state" portion of the solver should be doing. More so, where we get answers and such.

Reply:
Truncation *is* part of the fixed state -- it is a "deep property". i.e., if you are solving goal A, and it has a bunch of sub-goals, and some of them get truncated, the results you get for A are affected.